components/server/src/main/java/com/hotels/styx/server/netty/connectors/HttpPipelineHandler.java

```diff
@@ -667,7 +668,7 @@ private void cancelSubscription() {
         private HttpErrorStatusListener httpErrorStatusListener = IGNORE_ERROR_STATUS;
         private RequestProgressListener progressListener = IGNORE_REQUEST_PROGRESS;
         private HttpResponseWriterFactory responseWriterFactory = HttpResponseWriter::new;
-        private Supplier<CentralisedMetrics> metricsSupplier = () -> new CentralisedMetrics(new CompositeMeterRegistry());
+        private Supplier<CentralisedMetrics> metricsSupplier = () -> new CentralisedMetrics(new MicrometerRegistry(new CompositeMeterRegistry()));
```

Comment:
I know you didn't add this functionality,  but I'm not sure why we need a metrics supplier and can't just set metrics directly?

Reply:
I can't see any reason. I will see if I can simplify this.